aws-datastore/src/main/java/com/amplifyframework/datastore/syncengine/OutboxMutationEvent.java

```diff
@@ -31,7 +32,7 @@
  * and {@link DataStoreChannelEventName#OUTBOX_MUTATION_PROCESSED} event.
  * @param <M> The class type of the model in the mutation outbox.
  */
-public final class OutboxMutationEvent<M extends Model> {
+public final class OutboxMutationEvent<M extends Model> implements HubEvent.Data<OutboxMutationEvent<M>> {
```

Comment:
Let's either make _all_ of the payloads do this, or none. Can we save this for a follow up PR where we switch over all payload structures at once?

Reply:
this event is from an older PR and I probably shouldn't have made this change in this PR since it is unrelated. But yea, that was what I was going for (to make every payload implement `HubEvent.Data`).

Reply:
Addressed in [this commit](https://github.com/aws-amplify/amplify-android/pull/976/commits/01ee3ee914060394f95ac602fa3b84a3c4284fc3) (Needs review).